### PR TITLE
test: improve _fix_broken_tool_use test coverage

### LINKED ISSUE #2028: Test: _fix_broken_tool_use tests don't guard against in-place mutation; missing coverage for insert-induced edge case
## Summary

While reviewing #2026 I found two test quality issues in `tests/strands/session/test_repository_session_manager.py` for `_fix_broken_tool_use`.

### 1. `test_fix_broken_tool_use_ignores_last_message` doesn't actually guard against mutation

The test asserts:

```python
fixed_messages = session_manager._fix_broken_tool_use(messages)
assert fixed_messages == messages
```

Since `_fix_broken_tool_use` mutates the list **in-place** and returns the **same object**, `fixed_messages is messages` is always `True`. This means `fixed_messages == messages` passes regardless of whether the method appended anything. The test would still pass even after applying the fix from #2026.

**Fix:** Use `copy.deepcopy(messages)` to snapshot the original before calling the method, and compare against the snapshot.

### 2. No test for `insert()`-induced last-message skip

When consecutive orphaned `toolUse` messages exist without a trailing non-toolUse message, `insert()` at [L306](https://github.com/strands-agents/sdk-python/blob/main/src/strands/session/repository_session_manager.py#L306) shifts subsequent messages, causing the final `toolUse` to become the new "last message" mid-iteration. The `index + 1 < len(messages)` guard then skips it.

Example:
```python
messages = [
    {"role": "assistant", "content": [{"toolUse": {"toolUseId": "A", ...}}]},
    {"role": "assistant", "content": [{"toolUse": {"toolUseId": "B", ...}}]},
]
```

1. `index=0`: toolUse(A) → `insert(1, toolResult(A))` → list becomes `[A, resultA, B]` (len=3)
2. `index=2`: toolUse(B) → `index+1=3 < 3` is `False` → **skipped**

The `else` branch introduced in #2026 implicitly fixes this, but there's no test to pin the behavior.

### Proposed changes

- Apply `copy.deepcopy` to the existing last-message test
- Add a test for the consecutive orphaned toolUse scenario (insert-induced skip)
- Add a test for the single-message edge case (conversation is just one orphaned toolUse)

Related: #2025, #2026

### PR DESCRIPTION
**Pull request by beansandroasters**
## Summary

Improves test coverage for `_fix_broken_tool_use` in `RepositorySessionManager`, addressing test quality issues found while reviewing #2026.

- **Use `copy.deepcopy` in the existing last-message test** — the old assertion `fixed_messages == messages` compared the same object to itself (since the method mutates and returns the same list), so it always passed regardless of behavior changes
- **Add test for `insert()`-induced last-message skip** — when consecutive orphaned `toolUse` messages exist, `insert()` shifts the second into the "last message" position mid-iteration, causing it to be skipped by the `index + 1 < len(messages)` guard
- **Add test for single orphaned `toolUse`** — minimal edge case where the conversation consists of just one orphaned `toolUse` message

### Note

The two new tests **intentionally fail** on the current `main` branch — they document the existing bug that #2026 fixes. Once #2026 is merged, these tests will pass.

## Test plan

- [x] Existing 41 tests pass (no regressions)
- [x] Two new tests fail on current `main` (expected — they document the bug)
- [x] Two new tests pass after applying #2026 patch (verified locally)

Closes #2028
Related: #2025, #2026

### CODE REVIEW

File: tests/strands/session/test_repository_session_manager.py [hunk 1]
```diff
@@ -428,14 +434,75 @@ def test_fix_broken_tool_use_ignores_last_message(session_manager):
...
+    assert fixed_messages[1]["role"] == "user"
+    assert fixed_messages[1]["content"][0]["toolResult"]["toolUseId"] == "A"
+    assert fixed_messages[1]["content"][0]["toolResult"]["status"] == "error"
+
+    assert fixed_messages[2]["role"] == "assistant"
+    assert fixed_messages[2]["content"][0]["toolUse"]["toolUseId"] == "B"
+
+    assert fixed_messages[3]["role"] == "user"
+    assert fixed_messages[3]["content"][0]["toolResult"]["toolUseId"] == "B"
+    assert fixed_messages[3]["content"][0]["toolResult"]["status"] == "error"
```
Comment (zastrowm) [inline tests/strands/session/test_repository_session_manager.py:481]: Can we assert on the entire shape given that it's a UT?

File: tests/strands/session/test_repository_session_manager.py [hunk 2]
```diff
@@ -428,14 +434,75 @@ def test_fix_broken_tool_use_ignores_last_message(session_manager):
...
+        },
+    ]
+
+    fixed_messages = session_manager._fix_broken_tool_use(copy.deepcopy(messages))
+
+    assert len(fixed_messages) == 2
+    assert fixed_messages[1]["role"] == "user"
+    assert fixed_messages[1]["content"][0]["toolResult"]["toolUseId"] == "solo-123"
+    assert fixed_messages[1]["content"][0]["toolResult"]["status"] == "error"
+    assert fixed_messages[1]["content"][0]["toolResult"]["content"][0]["text"] == "Tool was interrupted."
```
Comment (zastrowm) [inline tests/strands/session/test_repository_session_manager.py:501]: Same as above; let's assert on the entire object

**Review by zastrowm — COMMENTED** [review]
Thanks; added some comments; I think you'll need to rebase as well
